mksnapshot.sh

```diff
@@ -10,40 +10,34 @@
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
 	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Where [branch] is the name of the branch to snapshot."
 	exit 1
 fi
-# VCS details
-module=squid3
-BZRROOT=${BZRROOT:-/bzr}
-
-# generate a tarball name from the branch ($1) note that trunk is at
-# /bzr/trunk, but we call it 3.HEAD for consistency with CVS (squid 2.x), and
-# branches are in /bzr/branches/ but we don't want 'branches/' in the tarball
-# name so we strip that.
-branchpath=${1:-trunk}
-tag=${2:-`basename $branchpath`}
+
+# generate a tarball name from the branch ($1)
+branch=${1:-master}
 startdir=${PWD}
 date=`env TZ=GMT date +%Y%m%d`
 
-tmpdir=${TMPDIR:-${PWD}}/${module}-${tag}-mksnapshot
+tmpdir=${TMPDIR:-${PWD}}/squid-${branch}-mksnapshot/
 
 rm -rf ${tmpdir}
 trap "echo FAIL-BUILD_${VERSION} ; rm -rf ${tmpdir}" 0
+mkdir ${tmpdir}
 
-rm -f ${tag}.out
-bzr export ${tmpdir} ${BZRROOT}/${module}/${branchpath} || exit 1
+rm -f ${branch}.out
+(git archive --format=tar ${branch} | tar -xC ${tmpdir}) || exit 1
 if [ ! -f ${tmpdir}/configure ] && [ -f ${tmpdir}/configure.ac ]; then
 	sh -c "cd ${tmpdir} && ./bootstrap.sh"
 fi
 if [ ! -f ${tmpdir}/configure ]; then
-	echo "ERROR! Tag ${tag} not found in ${module}"
+	echo "ERROR! Branch ${branch} not found."
 fi
 
 cd ${tmpdir}
-revision=`bzr revno ${BZRROOT}/${module}/${branchpath}`
+revision=`git rev-parse --short ${branch}`
 suffix="${date}-r${revision}"
-eval `grep "^ *PACKAGE_VERSION=" configure | sed -e 's/-BZR//' | sed -e 's/PACKAGE_//'`
+eval `grep "^ *PACKAGE_VERSION=" configure | sed -e 's/-VCS//' | sed -e 's/PACKAGE_//'`
```

Comment:
This is one of the places where it would be better to extract the "version" information from git rather than essentially expect all git branches to change `configure.ac`. I understand that you may not want to change this part of the snapshot "architecture" now, but it would be a perfect opportunity to do so IMHO.